README.md

```diff
@@ -1,2 +1,22 @@
 # comp110-worksheet-C
-Base repository for COMP110 worksheet C
+Base repository for COMP110 worksheet C - Aidan Crossan
+
+**a)** The algorithm performs a search to see if there are duplicate items in the same list (e.g. to see if there are more than two people in a list who have the name 'Garry' for instance).
+
+**b)** The least efficient way of cycling through the list is to check each of the items in the list in pairs to see whether they match or not. This would be quadratic complexity O(n^2).
+
+**c)** i-1 will still be similar to n-1, it just means that the j loop will be one behind the i loop. 
+
+**d)** The two loops ran twice in the original version meaning it had to go through one list and then run it again to see if there was any overlap.
```

Comment:
There are still two loops in this algorithm -- the key is noticing that the inner loop runs for half the number of iterations on average